### PR TITLE
feat: database rename

### PR DESCRIPTION
**Pull request by carols10cents**
Depends on https://github.com/influxdata/influxdb_iox/pull/5435 and https://github.com/influxdata/influxdb_iox/pull/5470

** might cause downtime/errors!!! **

### CODE REVIEW

File: iox_catalog/migrations/20220819154515_sequencer_to_shard.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+-- Rename sequencer to shard
+ALTER TABLE IF EXISTS sequencer RENAME TO shard;
+ALTER TABLE IF EXISTS shard RENAME CONSTRAINT sequencer_unique TO shard_unique;
+ALTER TABLE IF EXISTS PARTITION RENAME sequencer_id TO shard_id;
```
Comment (mkmik) [inline iox_catalog/migrations/20220819154515_sequencer_to_shard.sql:4]: for uniformity perhaps lowercase `partition` ?

Comment (carols10cents) [reply iox_catalog/migrations/20220819154515_sequencer_to_shard.sql:4]: Will do-- was copypastaing from the original schema that had partition in all caps for some reason. I like uniformity too 👍🏻

**Comment by carols10cents** [comment]
Ok this is the database rename, I think it's ready to go but I'm scared

**Comment by NGA-TRAN** [comment]
As much as we want this to happen, I think we need to discuss when we have this to go. Let me start a discussion with the team first

**Review by domodwyer — COMMENTED** [review]
Ship it and bring the (temporary) errors :fire:

**Review by domodwyer — APPROVED** [review]

**Pull request merged**